src/table.ts

```diff
@@ -937,19 +958,30 @@ Please use the format 'prezzy' or '${instance.name}/tables/prezzy'.`);
             return;
           }
           if (
-            numRequestsMade <= maxRetries &&
+            numConsecutiveErrors < maxRetries &&
             RETRYABLE_STATUS_CODES.has(error.code)
           ) {
-            makeNewRequest();
+            const backOffSettings =
+              options.gaxOptions?.retry?.backoffSettings ||
+              DEFAULT_BACKOFF_SETTINGS;
+            const nextRetryDelay = getNextDelay(
+              numConsecutiveErrors,
+              backOffSettings
+            );
+            retryTimer = setTimeout(makeNewRequest, nextRetryDelay);
+            numConsecutiveErrors++;
           } else {
             userStream.emit('error', error);
           }
         })
+        .on('data', _ => {
+          numConsecutiveErrors = 0;
+        })
         .on('end', () => {
           activeRequestStream = null;
+          retryTimer = null;
```

Comment:
Do you need to unpipe here?

Reply:
Yeah, I don't think we need to set it again since we reset it to null in the beginning of a new request. 

I think it's unpiped at line 796? 

```
    userStream.end = () => {
      rowStream?.unpipe(userStream);
```